samples/bluetooth/alexa_gadget/src/gadgets/CMakeLists.txt

```diff
@@ -0,0 +1,206 @@
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover.Response/alexaDiscoveryDiscoverResponseEvent.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Discovery/Discover.Response/alexaDiscoveryDiscoverResponseEventPayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/SetIndicator/notificationsSetIndicatorDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/SetIndicator/notificationsSetIndicatorDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Notifications/ClearIndicator/notificationsClearIndicatorDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/SetAlert/alertsSetAlertDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alerts/DeleteAlert/alertsDeleteAlertDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.StateListener/StateUpdate/alexaGadgetStateListenerStateUpdateDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.SpeechData/Speechmarks/alexaGadgetSpeechDataSpeechmarksDirectivePayload.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirective.proto
+    ${ALEXA_EMBEDDED_PATH}/AlexaGadgetsProtobuf/Alexa.Gadget.MusicData/Tempo/alexaGadgetMusicDataTempoDirectivePayload.proto
+    ${CMAKE_CURRENT_SOURCE_DIR}/proto/custom_event.proto
+)
+
+# Populate list of expected .c files output from protobuf generation
+foreach(file ${PROTO_FILES})
+    get_filename_component(FILE_PATH ${file} DIRECTORY)
+    get_filename_component(FILE_BASE ${file} NAME_WE)
+    list(APPEND PROTOC_INC_DIRS "-I${FILE_PATH}")
+    list(APPEND PROTO_BYPRODUCTS "${PROTO_COMPILATION_DIR}/${FILE_BASE}.pb.c")
+endforeach()
+
+# Populate list of compilation commands for protoc compiler.
+# Protoc is a bit fiddly with working directory.
+# Each .proto file is compiled with file path as working dir.
+# All other .proto files are provided as includes, using absolute links.
+foreach(file ${PROTO_FILES})
+    get_filename_component(FILE_PATH ${file} DIRECTORY)
+    get_filename_component(FILE_NAME ${file} NAME)
+    set(PROTOC_ARGS "${PROTOC_INC_DIRS}")
+    # Target .proto file should not be used as an include
+    list(REMOVE_ITEM PROTOC_ARGS "-I${FILE_PATH}")
+    list(APPEND PROTOC_ARGS "-I.")
```

Comment:
why is `-I.` better than `-I${FILE_PATH}` when you are changing directory into `${FILE_PATH}` for compilation ?
To me, it seems like it should result in same include paths, although the order has changed because of `-I.` is appended to the list (thus becomes last).

Reply:
I had trouble getting the include paths and working directory right for the protoc compilation. Revisited this command list, and see I can remove both these lines without errors now.